go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -900,6 +919,7 @@ func (d *DoltSession) SwitchWorkingSet(
 		return err
 	}
 
+	// TODO: should this be an error if any database in the transaction is dirty, or just this one?
```

Comment:
seems like if any database has been marked dirty from the current transaction, we'd want to return an error, since we know we won't be able to commit across all the dirty dbs. If someone doesn't care about the data changes to another db, that could be annoying to block them, but still seems like a reasonable behavior.

Reply:
Yeah, I'll make this concrete in my next PR